storage/cache/subtree_cache.go

```diff
@@ -313,29 +265,26 @@ func (s *SubtreeCache) getNodeHash(id storage.NodeID, getSubtree GetSubtreeFunc)
 		glog.Infof("cache: getNodeHash(path=%x, prefixLen=%d) {", id.Path, id.PrefixLenBits)
 	}
 
-	sInfo := s.stratumInfoForNodeID(id)
-	prefixKey := id.PrefixAsKey(sInfo.prefixBytes)
+	prefixID, sx := s.layout.Split(id)
+	prefixKey := string(prefixID.Path)
```

Comment:
Again I think it would be an improvement for the layout to return the key + 1 instance below.

Reply:
See above.